src/NUnitFramework/tests/Attributes/OneTimeSetUpTearDownTests.cs

```diff
@@ -372,6 +372,34 @@ public void DisposeCalledOnceWhenFixtureImplementsIDisposableAndHasTestCases()
             TestBuilder.RunTestFixture(fixture);
             Assert.That(fixture.DisposeCalled, Is.EqualTo(1));
         }
+
+        [Test]
+        public void AsyncDisposeCalledOnceWhenFixtureImplementsIAsyncDisposable()
+        {
+            var fixture = new AsyncDisposableFixture();
+            TestBuilder.RunTestFixture(fixture);
+            Assert.That(fixture.DisposeCalled, Is.EqualTo(1));
+            Assert.That(fixture.Actions, Is.EqualTo(new object[] { "OneTimeSetUp", "OneTimeTearDown", "DisposeAsync" }));
```

Comment:
I'm not a fan of hard-coded strings. They are brittle in case they are renamed in the fixture called.
My preference would be `nameof(AsyncDisposableFixture.OneTimeSetup)` etc.
I know these strings are all over nunit test code, but maybe we cannot introduce more.